lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -4105,6 +4106,20 @@ uint ThreadContext::GetRandomNumber()
 #endif
 }
 
+BOOL ThreadContext::GetAndClearIsScriptContextCloseGCPending()
+{
+    // Pretend that there is no GC pending if script code is still executing
+    // In that case, let the regular GC heuristics kick in
+    if (this->callRootLevel > 0)
```

Comment:
Instead of callRootLevel == 0 check. change it to add a CollectionFlags CollectOverride_CheckScriptcontextClose to CollectOnScriptExit to indicate whether we need to check this flag?